### ISSUE TITLE
Test case RD-179: Snapshot doesn't seem to respect the settings in a locked deployment profile

### ISSUE THREAD

**Issue by gunamata** [rancher-sandbox/rancher-desktop#6280]
**QASE Id:** RD-179
**1.12 RC4 build:** https://github.com/rancher-sandbox/rancher-desktop/actions/runs/7413086519

Snapshot restore doesn't seem to respect the settings in a locked deployment profile as stated in the test case. Logs from my machine at this state:

[RD-179-Step-5.zip](https://github.com/rancher-sandbox/rancher-desktop/files/13847666/RD-179-Step-5.zip)

**Comment by mook-as** [comment]
Did you restart the application after creating the locked deployment profile? Those are only applied on application start, and the test case instructions missed that part and they don't work.  The test case hasn't been fixed as of when I last checked.

**Comment by gunamata** [comment]
Yes, After the application restart, it works as described in the testcase. I will keep the ticket open till we fix the testcase instructions.

**Comment by jandubois** [comment]
I think we should keep the testcase as-is, but update the software.

I don't consider the current behaviour a bug or a vulnerability. It is not really different from the user keeping Rancher Desktop running while the locked profile is being updated (e.g. pushed centrally).

We don't enforce the new profile right away, which may require restarting the VM and could cause data loss. Instead we apply the new profile the next time the application starts.

So far this does not involve restoring a snapshot at all.

When we restore a snapshot via the GUI we also don't re-read the updated (or new) profiles. This is not fundamentally different from the behaviour above.

However, we easily can do this, as we have to stop and restart the VM anyways, so there is a perfect point to refresh the profile from disk.

So I think we should do this, so restoring a snapshot works implicitly the same as if the user stopped and restarted the application.
